src/ray/object_manager/ownership_based_object_directory.cc

```diff
@@ -121,16 +137,32 @@ ray::Status OwnershipBasedObjectDirectory::ReportObjectRemoved(
 void OwnershipBasedObjectDirectory::SubscriptionCallback(
     ObjectID object_id, WorkerID worker_id, Status status,
     const rpc::GetObjectLocationsOwnerReply &reply) {
+  // Objects are added to this map in SubscribeObjectLocations.
   auto it = listeners_.find(object_id);
+  // Do nothing for objects we are not listening for.
   if (it == listeners_.end()) {
     return;
   }
+  std::unordered_set<NodeID> node_ids;
+
+  if (!status.ok()) {
+    RAY_LOG(INFO) << "Worker " << worker_id << " failed to return location updates to "
+                  << "subscribers  for " << object_id << ": " << status.ToString()
+                  << ", assuming that the object was evicted.";
+    auto callbacks = it->second.callbacks;
```

Comment:
```suggestion
    auto &callbacks = it->second.callbacks;
```

Reply:
I think that we should still make a copy of the callbacks here so any unsubscribing calls won't interrupt our looping over the callbacks, right? I'll add a comment indicating as much, let me know if I'm wrong there.